dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/ImpersonateUserController.java

```diff
@@ -130,12 +129,23 @@ public ImpersonateUserResponse impersonateUser(
     }
   }
 
+  private void validateReq(HttpServletRequest request) throws ForbiddenException {
+    boolean enabled = config.isEnabled(ConfigurationKey.SWITCH_USER_FEATURE_ENABLED);
+    if (!enabled) {
+      throw new ForbiddenException("Forbidden, user not allowed to impersonate user");
+    }
+    if (!hasAllowListedIp(request.getRemoteAddr())) {
+      throw new ForbiddenException("Forbidden, user not allowed to impersonate user");
```

Comment:
could these `if` checks be combined if the outcome is exactly the same?
or, if the allow listed IP would have a different error message, then this would make more sense.